Communitter/src/main/java/com/communitter/api/service/PostService.java

```diff
@@ -58,6 +49,18 @@ public String deletePost(Long id){
         return "Post deleted";
     }
 
+    @Transactional
+    public PostVote votePost(Long id, boolean isUpvote){
+        User author= (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        Post postToVote =postRepository.findById(id).orElseThrow(()->new NoSuchElementException("Post does not exist"));
+        PostVote postVote = new PostVote(id, isUpvote, postToVote, author);
+        return postVoteRepository.save(postVote);
+    }
+
+    public Long getVoteCount(Long id){
+        return postVoteRepository.countVotesForPost(id, true) - postVoteRepository.countVotesForPost(id, false);
```

Comment:
I think handling this part in the frontend would be easier for you.

Reply:
Can you give details about whats is in your mind? If it is getting all postvotes to frontend it would unnecessarily increase the throughtput between client and server in my opinion.

Reply:
@Ozmus You can create a count state, every time someone click on upvote and send request to backend, you can update the state on frontend